docs/design/disable-add-in-commands.md

```diff
@@ -0,0 +1,179 @@
+
+Second, assign handlers. This is commonly done in the **Office.onReady** method as in the following example which assigns handlers (created in a later step) to the **onActivated** and **onDeactivated** events of all the charts in the worksheet.
+
+```javascript
+Office.onReady(async () => {
+    await Excel.run(context => {
+        var charts = context.workbook.worksheets
+            .getActiveWorksheet()
+            .charts;
+        charts.onActivated.add(enableChartFormat);
+        charts.onDeactivated.add(disableChartFormat);
+        return context.sync();
+    });
+});
+```
+
+Third, define the `enableChartFormat` handler. The following is a simple example, but see **Best practice: Test for control status errors** below for a more robust way of changing a control's status.
+
+```javascript
+function enableChartFormat() {
+    OfficeRuntime.ui.getRibbon()
+        .then(function (ribbon) {
+            var button = {id: "ChartFormatButton", enabled: true};
+            var parentTab = {id: "CustomChartTab", controls: [button]};
+            var ribbonUpdater = {tabs: [parentTab]};
+            await ribbon.requestUpdate(ribbonUpdater);
+        });
+}
+```
+
+Fourth, define the `disableChartFormat` handler. It would be identical to `enableChartFormat` except that the **enabled** property of the button object would be set to `false`.
+
+## Best practice: Test for control status errors
+
+In some circumstances, the ribbon does not repaint after `requestUpdate` is called, so the control's clickable status does not change. For this reason it is a best practice for the add-in to keep track of the status of its controls. The add-in should conform to these rules:
+
+1. Whenever `requestUpdate` is called, the code should record the intended state of the custom buttons and menu items.
+2. When a custom control is clicked, the first code in the handler that runs, should check to see if the button should have been clickable. If shouldn't have been, the code should report or log an error and try again to set the buttons to the intended state.
+
+The following example shows an example of a function that disables a button and records the button's status. About this code, note:
+
+- `chartFormatButtonEnabled` is a global boolean variable that is initialized to the same value as the [Enabled](/office/dev/add-ins/reference/manifest/enabled) element for the button in the manifest.
+
+```javascript
+function disableChartFormat() {
+    OfficeRuntime.ui.getRibbon()
+        .then(function (ribbon) {
+            var button = {id: "ChartFormatButton", enabled: false};
+            var parentTab = {id: "CustomChartTab", controls: [button]};
+            var ribbonUpdater = {tabs: [parentTab]};
+            await ribbon.requestUpdate(ribbonUpdater);
+
+            chartFormatButtonEnabled = false;
+        });
+}
+
+```
+
+The following example shows how the button's handler tests for an incorrect state of the button. About this code, note:
```

Comment:
Could the note be rewritten as "Note that `reportError` is a function that shows or logs an error." ? It just seems weird to me to have a single bulletpoint list.